src/Service.Host/client/src/components/salesPackages/SalesPackage.js

```diff
@@ -0,0 +1,275 @@
+import React, { Fragment, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Grid, Button, Table, TableBody, TableHead, TableRow, TableCell } from '@material-ui/core';
+import AddIcon from '@material-ui/icons/Add';
+import {
+    SaveBackCancelButtons,
+    InputField,
+    Loading,
+    Title,
+    ErrorCard,
+    SnackbarMessage
+} from '@linn-it/linn-form-components-library';
+import Page from '../../containers/Page';
+
+function SalesPackage({
+    item,
+    editStatus,
+    setEditStatus,
+    errorMessage,
+    itemId,
+    update,
+    history,
+    add,
+    loading,
+    snackbarVisible,
+    setSnackbarVisible
+}) {
+    const [salesPackage, setSalesPackage] = useState({});
+    const [prevSalesPackage, setPrevSalesPackage] = useState({});
+    const [showAddRow, setShowAddRow] = useState(false);
+    const [newElement, setNewElement] = useState({});
+
+    const creating = () => editStatus === 'create';
+    const editing = () => editStatus === 'edit';
+    const viewing = () => editStatus === 'view';
+    const cursor = {
+        cursor: 'pointer'
+    };
+    useEffect(() => {
+        if (!creating() && item !== prevSalesPackage) {
+            setSalesPackage(item);
+            setPrevSalesPackage(item);
+        }
+    });
+
+    const showFieldsToAddElement = () => {
+        setShowAddRow(true);
+    };
+
+    const inputInvalid = () => !salesPackage.salesPackageId;
+
+    const handleSaveClick = () => {
+        if (editing()) {
+            if (newElement.elementType) {
+                const { elements } = salesPackage;
+                elements.push(newElement);
+                setSalesPackage({ ...salesPackage, elements });
+            }
+            update(itemId, salesPackage);
+            setEditStatus('view');
+            setNewElement({});
+            setShowAddRow(false);
+        } else if (creating()) {
+            add(salesPackage);
+        }
+    };
+
+    const handleCancelClick = () => {
+        setSalesPackage(item);
+        setEditStatus('view');
+    };
+
+    const handleBackClick = () => {
+        history.push('/products/maint/sales-packages');
+    };
+
+    const handleFieldChange = (propertyName, newValue) => {
+        setEditStatus('edit');
+        setSalesPackage({ ...salesPackage, [propertyName]: newValue });
+    };
+
+    const handleNewElement = (propertyName, newValue) => {
+        setEditStatus('edit');
+        setNewElement({ ...newElement, [propertyName]: newValue });
+    };
+
+    const handleElementChange = (propertyName, newValue) => {
+        setEditStatus('edit');
+        const splitIndex = propertyName.indexOf(',');
+        const index = propertyName.slice(0, splitIndex);
+        const prop = propertyName.slice(splitIndex + 1);
+        const { elements } = salesPackage;
+        elements[index][prop] = newValue;
```

Comment:
could you pass in index and prop into this function instead of using split etc? Not sure it that would be better or not...

Reply:
So the index of the element is passed in via propertyName {`${index},propertyName`}.. the split index might change depending on the number.. don't know if that makes sense!